overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -5004,12 +5004,12 @@ plugs:
 			RealName: "foo-missing-deps",
 		})
 		snapstate.Set(s.state, info.InstanceName(), &snapstate.SnapState{
-			SnapType: string(info.Type()),
-			Active:   true,
-			Sequence: []*snap.SideInfo{&info.SideInfo},
-			Current:  info.Revision,
+			SnapType:        string(info.Type()),
+			Active:          true,
+			Sequence:        []*snap.SideInfo{&info.SideInfo},
+			Current:         info.Revision,
+			TrackingChannel: "latest/stable",
```

Comment:
This test-case was taking the `else` branch that I mentioned in a previous comment. I believe the intention of the test was that the snap was unchanged. Adding the tracking channel makes the snap appear as unchanged.

Reply:
that's fair but we still need to have a test about  that else though, as the old behavior was correct and needs to be tested

Reply:
That test tests a channel switch for the base snap, which is handled differently. I'll make a new one that exercises this functionality.